internal/tester/data.go

```diff
@@ -293,6 +299,24 @@ func (t *DataTester) StartPeriodicLogger(
 	return ctx.Err()
 }
 
+// WatchEndConditions starts go routines to watch the end conditions
+func (t *DataTester) WatchEndConditions(
+	ctx context.Context,
+	config *configuration.Configuration,
+) error {
```

Comment:
nit: with `EndCondition` as null, you can just return (when it is set to a pointer)